src/AstSemanticChecker.cpp

```diff
@@ -445,6 +469,30 @@ void AstSemanticChecker::checkAggregator(
         report.addError("Unsupported nested aggregate", inner->getSrcLoc());
     }
 
+    AstClause clauseAggregator;
+    // check whether this aggregate 'aggregator' is mutually dependent with
+    // another aggregate contained in this rule. This is disallowed
+    visitDepthFirst(program, [&](const AstLiteral& literal) {
+        visitDepthFirst(literal, [&](const AstAggregator& agg) {
+            if (agg == aggregator) {
```

Comment:
This is a reasonable way to show the logic, but if you do
```
if (agg != aggregator) {
    return;
}
```
then there is less indentation and readability slightly improves.

Reply:
I can't do it, because if the first aggregate that it iterates over isn't the agg, it will stop the iteration all together and won't reach the one i'm looking for